woocommerce/woocommerce-gutenberg-products-block
# improve responsiveness when setting cart item quantities

Fixes #1850

This PR improves how cart line item quantities work, allowing the user to increase/decrease quantity, and defers the server update to reduce unnecessary network requests.

- The generator action for persisting quantity (`changeCartItemQuantity`) is now simpler. It doesn't set the item as pending, so the UI stays enabled and available when API requests are in progress. 
- Added local state in `CartLineItemRow` component so the quantity in the UI updates immediately.
- The UI quantity value is debounced and then persisted when stable (via `changeCartItemQuantity`).

### How to test the changes in this Pull Request:

1. Add cart block to a page and publish, add some stuff to cart.
2. Change item quantity using quantity picker - plus/minus buttons and text edit.
3. Confirm that quantities correctly persist and the UI is responsive to user input at all times.
